chrome-extension/content.js

```diff
@@ -162,26 +181,31 @@ async function getCategoryDropdown(userId) {
       categoryDropdown.options.add(newOption);
     });
   } catch (error) {
-    // TODO: Handle this error?
+    // TODO: Handle error
   }
   return categoryDropdown;
 }
 
 /**
  * Add new item to databse
  */
-async function addItem() {
-  newItem = {
+async function addItem(mongoId) {
+  const newItem = {
     sourceLink: window.location.toString(),
     placesId: 'something', // TODO: get actual placesId
-    userId: currentUserId, // TODO: get actual userID
     highlight: document.getElementById('highlight').value,
     comment: document.getElementById('comment').value,
   };
 
-  const selectedCategories = getSelectedOptions('categoryDropdown');
+  if (mongoId) {
+    newItem.userId = mongoId;
+  } else {
+    alert('Please sign into Chrome to save your clipping');
+  }
```

Comment:
It looks like the if the user wasn't signed in, this would still try to send the request to add an item. Consider returning if the user isn't signed out?

Reply:
yes thank you for catching! I put return;